exporters/dynamicconfig/confignotifier.go

```diff
@@ -167,15 +178,23 @@ func (n *ConfigNotifier) checkChanges(ch chan struct{}) {
 		case <-n.ticker.C():
 			newConfig := readConfig(n.configHost)
 
-			n.lock.Lock()
 			if !n.config.equals(newConfig) {
+				n.lock.Lock()
 				n.config = newConfig
 
+				// To prevent lock starvation, we will copy the list of subscribers, then update
+				subscribed_copy := make([]ConfigWatcher, len(n.subscribed))
+				i := 0
+				for watcher := range n.subscribed {
+					subscribed_copy[i] = watcher
+					i++
```

Comment:
for i, watcher in enumerate(xxx)?

Reply:
I don't think there was a better way to do it, in terms of getting keys from a map.

It is cleaner to just copy over a new map, though copying just the keys into a list might be slightly more performant. Let me know your thoughts.